pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseMultipleSegmentsConversionExecutor.java

```diff
@@ -107,14 +107,31 @@ protected abstract List<SegmentConversionResult> convert(PinotTaskConfig pinotTa
 
   /**
    * Pre processing operations to be done at the beginning of task execution
+   *
+   * The default implementation checks whether all segments to process exist in the table, if not, terminate early to
+   * avoid wasting computing resources.
    */
-  protected void preProcess(PinotTaskConfig pinotTaskConfig) {
+  protected void preProcess(PinotTaskConfig pinotTaskConfig) throws Exception {
+    Map<String, String> configs = pinotTaskConfig.getConfigs();
+    String tableNameWithType = configs.get(MinionConstants.TABLE_NAME_KEY);
+    String inputSegmentNames = configs.get(MinionConstants.SEGMENT_NAME_KEY);
+    String uploadURL = configs.get(MinionConstants.UPLOAD_URL_KEY);
+    AuthProvider authProvider = AuthProviderUtils.makeAuthProvider(configs.get(MinionConstants.AUTH_TOKEN));
+    Set<String> segmentNamesForTable = SegmentConversionUtils.getSegmentNamesForTable(tableNameWithType,
+        FileUploadDownloadClient.extractBaseURI(new URI(uploadURL)), authProvider);
+    Set<String> nonExistingSegmentNames =
+        new HashSet<>(Arrays.asList(inputSegmentNames.split(MinionConstants.SEGMENT_NAME_SEPARATOR)));
+    nonExistingSegmentNames.removeAll(segmentNamesForTable);
+    if (!CollectionUtils.isEmpty(nonExistingSegmentNames)) {
+      throw new RuntimeException(String.format("table: %s does have the following segments to process: %s",
+          tableNameWithType, nonExistingSegmentNames));
+    }
   }
 
   /**
    * Post processing operations to be done before exiting a successful task execution
    */
-  protected void postProcess(PinotTaskConfig pinotTaskConfig) {
+  protected void postProcess(PinotTaskConfig pinotTaskConfig) throws Exception {
   }
```

Comment:
is this change needed?

Reply:
The purpose is to make it consistent with `preUploadSegmentsm`,  `postUploadSegments` and `preprocess`